### ISSUE TITLE
*: migrate the creation of internal executor to the new interfaces

### ISSUE THREAD

**Issue by ZhouXing19** [cockroachdb/cockroach#91004]
TL;DR:
- To run multiple SQL statements and/or DDL statements in a transactional manner, use [sql.planner.WithInternalExecutor()](https://github.com/cockroachdb/cockroach/blob/29c8e6e0c73046a9e8cd52cfd67bebecec9a91e9/pkg/sql/planner.go#L999-L1008), [sqlutil.InternalExecutorFactory.TxnWithExecutor()](https://github.com/cockroachdb/cockroach/blob/3a22cc763dc828ca07d0162ae014f1ae01b91e12/pkg/sql/sqlutil/internal_executor.go#L221-L232), or  [descs.CollectionFactory.TxnWithExecutor()](https://github.com/cockroachdb/cockroach/blob/3a22cc763dc828ca07d0162ae014f1ae01b91e12/pkg/sql/catalog/descs/factory.go#L52-L64). ([example](https://github.com/cockroachdb/cockroach/blob/3a22cc763dc828ca07d0162ae014f1ae01b91e12/pkg/cloud/userfile/filetable/file_table_read_writer.go#L248-L276))
- To run other single statements under a planner context, use query functions under planner (e.g. [sql.planner.QueryRowEx()](https://github.com/cockroachdb/cockroach/blob/2541f2e5d4d0aea29455ccbfd394ea63cc751eb7/pkg/sql/planner.go#L916) or [sql.planner.WithInternalExecutor()](https://github.com/cockroachdb/cockroach/blob/ae38d903c527fdd85a0183b6245d191e51d79578/pkg/sql/planner.go#L1003))
- Otherwise, use sqlutil.InternalExecutorFactory.MakeInternalExecutorWithoutTxn().

We'd like to migrate the existing usages of internal executor to the interfaces above. Once these are all using the new interfaces, we will remove the txn parameter in the query functions (e.g. ie.ExecEx()) and use ie.extraTxnState.txn.

## Motivation
The previous design of the internal executor is either incorrect or inefficient when running with an outer txn, as it always creates a new set of txn-related metadata (such as descriptor collection and schema change job records) for its child conn executor. This is not intuitive, since it violates a rather deep principle that these metadata and a txn have a 1:1 relationship. The lack of such coupling has caused issues such as [filetable corruption](https://github.com/cockroachdb/cockroach/issues/76764), unnecessary rescanning of the descriptors, etc.

## New interfaces
The new internal executor interfaces (implemented in [#82477](https://github.com/cockroachdb/cockroach/pull/82477), [#86427](https://github.com/cockroachdb/cockroach/pull/86427), and parts of [#87067](https://github.com/cockroachdb/cockroach/pull/87067)) are meant to solve this problem. They ensure that the internal executor, if used with an outer txn, is bound with the correct metadata, which will be passed to the child conn executor. When the internal executor is closed, the sql transaction will be committed.

Epic: [CRDB-19135](https://cockroachlabs.atlassian.net/browse/CRDB-19135)

Jira issue: CRDB-21069

Epic CRDB-19135

**Comment by rimadeodhar** [comment]
Synced with @ajwerner who confirmed that this issue has been completed by PR https://github.com/cockroachdb/cockroach/pull/93218. I'm closing out this issue.
